views/userAllPackage.ejs

```diff
@@ -0,0 +1,44 @@
+<div class="container" id="search">
+<% if(packages.length >0){%>
+<form class="form my-3" action="#" method="GET">
+	<div class="input-group">
+    <input class="form-control form-control-lg" type="text" placeholder="Search Swift Package Registry" id="term" name="term">
+    <span class="input-group-append">
+    	<button type="submit" class="btn btn-primary" disabled>Search</button>
+    	</span>
+    </input>
+    </div>
+</form>
+<h2>Swift Packages</h2>
+<table class="table table-striped table-hover table-responsive">
+	<thead class="thead-dark">
+		<tr>
+		<th scope="col">Repository</th>
+		<th scope="col">Description</th>
+		</tr>
+	</thead>
+	<tbody>
+  <% packages.forEach(function(package){ %>
+	<tr>
+		<td><a href="/<%= package.info.full_name %>">
+			<% if (package.description) { %>
+				<%= package.description.name %>
+			<% } else { %>
+				<%= package.info.name %>
+			<% } %>
+		</a></td>
+
+		<td><%= package.error || package.info.description %></td>
+	</tr>
+  <% }); %>
+  </tbody>
+</table>
+
+<%} else{ %>
+	<div>
+		<h1 class="text-center text-danger" >404</h1>
```

Comment:
Can the route handle this?

res.render('404')

Reply:
Yes, It is handling. if(packages.length >0) does not true then below code will be executed. It's means the user's has not packages added so the user is not exist. i.e( http://localhost:3000/fastlanessafs). in this case **fastlanessafs** does not exist. 
<%} else{ %>
	<div>
		<h1 class="text-center text-danger" >404</h1>
		<h4 class="text-center"><strong>oOPs! </strong>PAGE NOT FOUND</4>
	</div
<%}%>

Reply:
If the user doesn't exist then it should 404 in the route code then - not the template.